pop-benchmarker/src/main/java/com/microsoft/testing/popbenchmarker/MainActivity.java

```diff
@@ -0,0 +1,385 @@
+import android.widget.TextView;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.crypto.RSASSASigner;
+import com.nimbusds.jose.crypto.impl.RSAKeyUtils;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.Semaphore;
+
+import static com.microsoft.testing.popbenchmarker.PopUtils.ANDROID_KEYSTORE;
+import static com.microsoft.testing.popbenchmarker.PopUtils.KEYSTORE_ALIAS;
+import static com.microsoft.testing.popbenchmarker.PopUtils.getInitializedRsaKeyPairGenerator;
+import static com.microsoft.testing.popbenchmarker.PopUtils.isInsideSecureHardware;
+
+
+public class MainActivity extends AppCompatActivity {
+
+    private MainThreadMarshaller<String> mThreadMarshaller = new MainThreadMarshaller<>();
+
+    private List<Long> mKeyGenerationTimings = new ArrayList<>();
+    private List<Long> mKeyLoadTimings = new ArrayList<>();
+    private List<Long> mSigningTimings = new ArrayList<>();
+    //private boolean mAlreadyExecuted = false;
+
+    private TextView
+            mTv_Manufacturer,
+            mTv_Model,
+            mTv_OsVer,
+            mTv_ApiLevel,
+            mTv_KeyGen,
+            mTv_KeyLoad,
+            mTv_Signing,
+            mTv_HardwareIsolated;
+
+    private Button mBtn_Restart;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        initializeViews();
+        setText(mTv_Manufacturer, Build.MANUFACTURER);
+        setText(mTv_Model, Build.MODEL);
+        setText(mTv_OsVer, Build.VERSION.RELEASE);
+        setText(mTv_ApiLevel, Build.VERSION.SDK_INT);
+
+        executeBenchmarks();
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+    }
+
+    private void executeBenchmarks() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                final int iterations = 20;
+                int thisIteration = iterations;
+
+                while (thisIteration > 1) {
+                    final int finalThisIteration = thisIteration;
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            mBtn_Restart.setText("Iteration: " + finalThisIteration + " of: " + iterations);
+                        }
+                    });
+                    populateViews();
+                    thisIteration--;
+                }
```

Comment:
Instead of going from 20 to 1, can we go from 1 to 20 instead? I think that will make it more easily readable (at least for me) because it looks a bit confusing when on line 103, we assign a plural to a singular. 

Also when printing we are first printing iteration 20 of 20 then going backwards. I think it might be easier to read them in order.

Reply:
UX isn't really the point here, I've changed the language on the button in 7c7d006b370cb973e13066f7fe2902ef66599ca7 to make this clearer. With this change, I'm going to leave as-is

Reply:
Cool, Just curious though, any particular reason to go with a `while` loop instead of a `for` loop? This seems more suited to a for loop to me.